apps/vc-api/src/vc-api/credentials/credentials.service.ts

```diff
@@ -2,93 +2,67 @@
-  challenge?: string;
-}
+import { CredoService } from '../../credo/credo.service';
+import {
+  ClaimFormat,
+  JsonTransformer,
+  W3cCredential,
+  W3cJsonLdSignCredentialOptions,
+  W3cJsonLdVerifiableCredential,
+  W3cJsonLdVerifiablePresentation,
+  W3cPresentation,
+  W3cSignPresentationOptions,
+  W3cVerifyCredentialOptions,
+  W3cVerifyCredentialResult,
+  W3cVerifyPresentationOptions
+} from '@credo-ts/core';
+import { AuthenticateDto } from './dtos/authenticate.dto';
+import { transformVerificationResult } from './utils/verification-result-transformer';
 
 /**
  * This service provide the VC-API operations
  * This encapsulates the use of Spruce DIDKit
  */
 @Injectable()
 export class CredentialsService implements CredentialVerifier {
-  constructor(private didService: DIDService, private keyService: KeyService) {}
+  constructor(private didService: DIDService, private credoService: CredoService) {}
 
   async issueCredential(issueDto: IssueCredentialDto): Promise<VerifiableCredentialDto> {
     const verificationMethod = await this.getVerificationMethodForDid(
       typeof issueDto.credential.issuer === 'string'
         ? issueDto.credential.issuer
         : issueDto.credential.issuer.id
     );
-    const key = await this.getKeyForVerificationMethod(verificationMethod.id);
-    const proofOptions = this.mapVcApiIssueOptionsToSpruceIssueOptions(
-      issueDto.options || ({} as IssueOptionsDto),
-      verificationMethod.id
-    );
-
-    return didKitExecutor<VerifiableCredentialDto>(
-      () =>
-        issueCredential(
-          JSON.stringify(issueDto.credential),
-          JSON.stringify(proofOptions),
-          JSON.stringify(key)
-        ),
-      'issueCredential'
-    );
+    const credentialOption: W3cJsonLdSignCredentialOptions = {
+      credential: W3cCredential.fromJson(issueDto.credential),
+      format: ClaimFormat.LdpVc,
+      proofType: 'Ed25519Signature2018',
+      verificationMethod: verificationMethod.id
+    };
+    const w3cVerifiableCredential =
+      await this.credoService.agent.w3cCredentials.signCredential<ClaimFormat.LdpVc>(credentialOption);
+    return w3cVerifiableCredential.toJson() as VerifiableCredentialDto;
   }
 
-  async verifyCredential(
-    vc: VerifiableCredentialDto,
-    options: VerifyOptionsDto
-  ): Promise<VerificationResultDto> {
-    const verifyOptions: ISpruceVerifyOptions = options;
-
-    return didKitExecutor<VerificationResultDto>(
-      () => verifyCredential(JSON.stringify(vc), JSON.stringify(verifyOptions)),
-      'verifyCredential'
-    );
+  async verifyCredential(vc: VerifiableCredentialDto): Promise<VerificationResultDto> {
+    const w3cVerifyCredentialOptions: W3cVerifyCredentialOptions<ClaimFormat.LdpVc> = {
+      credential: W3cJsonLdVerifiableCredential.fromJson(vc)
+    };
+    const verifyCredential: W3cVerifyCredentialResult =
+      await this.credoService.agent.w3cCredentials.verifyCredential(w3cVerifyCredentialOptions as any);
```

Comment:
Followup from https://github.com/impactility/vc-api/pull/2#discussion_r1766892578
We should disable the `any` linting error if we are purposefully doing it, IMO.
```suggestion
    // Using "any" until Credo types are fixed https://github.com/openwallet-foundation/credo-ts/issues/2043
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const verifyCredential: W3cVerifyCredentialResult =
      await this.credoService.agent.w3cCredentials.verifyCredential(w3cVerifyCredentialOptions as any);
```